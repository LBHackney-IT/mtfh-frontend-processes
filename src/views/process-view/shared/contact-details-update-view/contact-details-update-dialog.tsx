import React from "react";

import {
  Button,
  Center,
  Dialog,
  PageAnnouncementProvider,
  Spinner,
  Text,
} from "@mtfh/common/lib/components";
import { useErrorCodes } from "@mtfh/common/lib/hooks";
import { ContactDetailsView } from "@mtfh/personal-details";
import "./styles.scss";

export const ContactDetailsUpdateDialog = ({ isDialogOpen, setDialogOpen, personId }) => {
  const errorMessages = useErrorCodes();

  if (!errorMessages) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const close = () => setDialogOpen(false);

  return (
    <Dialog
      isOpen={isDialogOpen}
      onDismiss={close}
      title="Update contact details"
      className="lbh-dialog-contact-details"
    >
      <Text size="sm">
        The details will be instantly updated within MMH when you save the email
        address/phone number.
      </Text>
      <Text size="sm">
        Only the first phone and email address will be displayed with the MMH application
        e.g. for appointment bookings.
      </Text>
      <Text size="sm">
        You can 'Return to your application' once you have completed the updates. The new
        contact details will be displayed.
      </Text>
      <PageAnnouncementProvider sessionKey="editPerson">
        <ContactDetailsView
          providedPersonId={personId}
          onComplete={(_, hasUpdatedContactDetails: boolean) => {
            if (hasUpdatedContactDetails) {
              sessionStorage.setItem("person:heading", "Person Updated");
            }
          }}
          hideAddress
          isInADialog
        />
      </PageAnnouncementProvider>
      <Button
        data-testid="close-update-contact-details-modal"
        variant="primary"
        onClick={close}
      >
        Return to application
      </Button>
    </Dialog>
  );
};
