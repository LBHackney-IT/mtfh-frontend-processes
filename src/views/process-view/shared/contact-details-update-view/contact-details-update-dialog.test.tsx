import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved } from "@testing-library/react";

import { ContactDetailsUpdateDialog } from "./contact-details-update-dialog";

import * as errorMessages from "@mtfh/common/lib/hooks/use-error-codes";

const setDialogOpen = jest.fn();

beforeEach(() => {
  jest.resetModules();
  jest.spyOn(errorMessages, "useErrorCodes").mockReturnValue({});
});

test("it ContactDetailsUpdateDialog as expected when it is open", async () => {
  const { baseElement } = render(
    <ContactDetailsUpdateDialog
      isDialogOpen
      setDialogOpen={setDialogOpen}
      personId="85e030a6-3ecf-424e-b965-0c9b5c19f21e"
    />,
  );
  await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
  await expect(screen.findByText("Return to application")).resolves.toBeInTheDocument();
  expect(screen.queryByText("Next")).not.toBeInTheDocument();
  expect(screen.queryByText("Add a correspondence address")).not.toBeInTheDocument();
  await screen.findByText("Email addresses");
  expect(baseElement).toMatchSnapshot();
});
