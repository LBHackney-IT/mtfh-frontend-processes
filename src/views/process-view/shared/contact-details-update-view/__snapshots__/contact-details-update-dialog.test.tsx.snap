// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it ContactDetailsUpdateDialog as expected when it is open 1`] = `
<body>
  <div
    aria-hidden="true"
  />
  <reach-portal>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div>
        <div
          data-reach-dialog-overlay=""
        >
          <div
            aria-label="Update contact details"
            aria-modal="true"
            class="lbh-dialog lbh-dialog-contact-details"
            data-reach-dialog-content=""
            role="dialog"
            tabindex="-1"
          >
            <h2
              class="lbh-heading-h2 lbh-dialog__title"
            >
              Update contact details
            </h2>
            <button
              class="lbh-dialog__close"
              type="button"
            >
              <span
                class="govuk-visually-hidden"
              >
                Close
              </span>
              <svg
                fill="none"
                height="18"
                viewBox="0 0 13 13"
                width="18"
              >
                <path
                  d="M-0.0501709 1.36379L1.36404 -0.050415L12.6778 11.2633L11.2635 12.6775L-0.0501709 1.36379Z"
                  fill="#0B0C0C"
                />
                <path
                  d="M11.2635 -0.050293L12.6778 1.36392L1.36404 12.6776L-0.0501709 11.2634L11.2635 -0.050293Z"
                  fill="#0B0C0C"
                />
              </svg>
            </button>
            <p
              class="lbh-body-s"
            >
              The details will be instantly updated within MMH when you save the email address/phone number.
            </p>
            <p
              class="lbh-body-s"
            >
              Only the first phone and email address will be displayed with the MMH application e.g. for appointment bookings.
            </p>
            <p
              class="lbh-body-s"
            >
              You can 'Return to your application' once you have completed the updates. The new contact details will be displayed.
            </p>
            <fieldset
              class="govuk-fieldset lbh-fieldset"
            >
              <legend
                class="govuk-fieldset__legend govuk-fieldset__legend--m"
              >
                
              </legend>
              <div
                class="mtfh-fieldset__content"
              >
                <div
                  class="mtfh-contact-details-form"
                >
                  <button
                    class="govuk-button lbh-button"
                    data-testid="button-add-email"
                    type="button"
                  >
                    Add an email address
                  </button>
                </div>
                <div
                  class="mtfh-contact-details-form"
                >
                  <button
                    class="govuk-button lbh-button"
                    data-testid="button-add-phone"
                    type="button"
                  >
                    Add a phone number
                  </button>
                </div>
              </div>
            </fieldset>
            <button
              class="govuk-button lbh-button"
              data-testid="close-update-contact-details-modal"
              type="button"
            >
              Return to application
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </reach-portal>
</body>
`;
